var/spack/repos/builtin/packages/py-jaxlib/package.py

```diff
@@ -19,23 +19,43 @@ class PyJaxlib(PythonPackage, CudaPackage):
 
     license("Apache-2.0")
 
+    version("0.4.24", sha256="c4e6963c2c36f634a9a1765e476a1ed4e6c4a7954465ebf72e29f344c28ddc28")
     version("0.4.3", sha256="2104735dc22be2b105e5517bd5bc6ae97f40e8e9e54928cac1585c6112a3d910")
-    version("0.3.22", sha256="680a6f5265ba26d5515617a95ae47244005366f879a5c321782fde60f34e6d0d")
-    version("0.1.74", sha256="bbc78c7a4927012dcb1b7cd135c7521f782d7dad516a2401b56d3190f81afe35")
+    version(
+        "0.3.22",
+        sha256="680a6f5265ba26d5515617a95ae47244005366f879a5c321782fde60f34e6d0d",
+        deprecated=True,
+    )
+    version(
+        "0.1.74",
+        sha256="bbc78c7a4927012dcb1b7cd135c7521f782d7dad516a2401b56d3190f81afe35",
+        deprecated=True,
+    )
 
     variant("cuda", default=True, description="Build with CUDA")
 
     # jaxlib/setup.py
+    depends_on("python@3.9:", when="@0.4.14:", type=("build", "run"))
     depends_on("python@3.8:", when="@0.4:", type=("build", "run"))
     depends_on("py-setuptools", type="build")
+    depends_on("py-scipy@1.9:", when="@0.4.19:", type=("build", "run"))
+    depends_on("py-scipy@1.7:", when="@0.4.7:", type=("build", "run"))
+    depends_on("py-scipy@1.5:", type=("build", "run"))
+    depends_on("py-numpy@1.22:", when="@0.4.14:", type=("build", "run"))
+    depends_on("py-numpy@1.21:", when="@0.4.7:", type=("build", "run"))
     depends_on("py-numpy@1.20:", when="@0.3:", type=("build", "run"))
     depends_on("py-numpy@1.18:", type=("build", "run"))
-    depends_on("py-scipy@1.5:", type=("build", "run"))
+    depends_on("py-ml-dtypes@0.2:", when="@0.4.14:", type=("build", "run"))
+    depends_on("py-ml-dtypes@0.1:", when="@0.4.9:", type=("build", "run"))
+    depends_on("py-ml-dtypes@0.0.3:", when="@0.4.7:", type=("build", "run"))
 
     # .bazelversion
-    depends_on("bazel@5.1.1:5.9", when="@0.3:", type="build")
-    # https://github.com/google/jax/issues/8440
-    depends_on("bazel@4.1:4", when="@0.1", type="build")
+    depends_on("bazel@6.1.2", when="@0.4.11:", type="build")
+    depends_on("bazel@5.1.1", when="@0.3.7:0.4.10", type="build")
+    depends_on("bazel@5.1.0", when="@0.3.5", type="build")
+    depends_on("bazel@5.0.0", when="@0.3.0:0.3.2", type="build")
+    depends_on("bazel@4.2.1", when="@0.1.75:0.1.76", type="build")
+    depends_on("bazel@4.1.0", when="@0.1.70:0.1.74", type="build")
```

Comment:
This should be less controversial. We're going from major version pinning to patch version pinning. This is also how we do things in packages like TF. Now that Spack's concretizer allows the same build dep to appear at two different versions in the DAG, this shouldn't cause any troubles.